discovery-provider/src/api/v1/users.py

```diff
@@ -2227,11 +2250,48 @@ class FullSalesCount(Resource):
     def get(self, id, authed_user_id=None):
         decoded_id = decode_with_abort(id, full_ns)
         if decoded_id != authed_user_id:
-            full_ns.abort(403)
-            return
+            abort_forbidden(full_ns)
         args = purchases_and_sales_count_parser.parse_args()
         args = GetUSDCPurchasesCountArgs(
             seller_user_id=decoded_id,
         )
         count = get_usdc_purchases_count(args)
         return success_response(count)
+
+
+@full_ns.route("/<string:id>/sales/download")
+class FullSalesDownload(Resource):
+    @full_ns.doc(
+        id="Download Sales as CSV",
+        description="Downloads the sales the user has made as a CSV file",
+        params={"id": "A User ID"},
+    )
+    @auth_middleware()
+    def get(self, id, authed_user_id=None):
+        decoded_id = decode_with_abort(id, full_ns)
+        if decoded_id != authed_user_id:
+            abort_forbidden(full_ns)
+        args = DownloadSalesArgs(seller_user_id=decoded_id)
+        sales = download_sales(args)
+        response = Response(sales, content_type="text/csv")
+        response.headers["Content-Disposition"] = "attachment; filename=sales.csv"
+        return response
+
+
+@full_ns.route("/<string:id>/withdrawals/download")
+class GetUSDCTransactionHistory(Resource):
+    @full_ns.doc(
```

Comment:
We'll probably need some way to generate the Swagger response we need... looking into this now